Runtime/BacktraceClient.cs

```diff
@@ -824,10 +854,22 @@ private bool ShouldSendReport(Exception exception, List<string> attachmentPaths,
             {
                 return false;
             }
+
             //check rate limiting
             bool shouldProcess = _reportLimitWatcher.WatchReport(new DateTime().Timestamp());
             if (shouldProcess)
             {
+                // This condition checks if we should send exception from current thread
+                // if comparision result confirm that we're trying to send an exception from different
+                // thread than main, we should add the exception object to the exception list 
+                // and let update method send data to Backtrace.
+                if (Thread.CurrentThread.ManagedThreadId != _current.ManagedThreadId)
+                {
+                    var report = new BacktraceReport(exception, attributes, attachmentPaths);
```

Comment:
This seems to be repeated 3 times. Can we pull this into a function?

Reply:
Not really because of different parameters that this function require and cost of BacktraceReport creation

Reply:
Okay I see. Since this gets called in Update method we need to consider cost of BacktraceReport creation.